scripts/home_allocation.py

```diff
@@ -0,0 +1,138 @@
+import argparse
+import os
+from typing import Literal
+
+import geopandas as gpd
+import polars as pl
+from context import nhs
+from fiona.drvsupport import supported_drivers
+from loguru import logger
+
+from nhs.config import simulation_config
+
+geography = nhs.data.geography
+read_parquet = nhs.data.read_parquet
+join_coords_with_area = nhs.data.join_coords_with_area
+to_geo_dataframe = nhs.data.to_geo_dataframe
+read_shapefile = nhs.data.read_shapefile
+join_coords_with_area = nhs.data.join_coords_with_area
+join_census_with_coords = nhs.data.join_census_with_coords
+randomly_assign_census_features = nhs.data.randomly_assign_census_features
+sample_census_feature = nhs.data.sample_census_feature
+
+
+def main(
+    config_path: str,
+    gnaf_fname: str,
+    census_fname: str,
+    out_fname: str,
+    strategy: Literal["join_nearest", "filter"] | None,
+) -> None:
+    # Required for fiona - reads shapefiles
+    supported_drivers["ESRI Shapefile"] = "rw"
+
+    try:
+        data_config = nhs.config.data_config(config_path)
+        logger_config = nhs.config.logger_config(config_path)
+        simulation_config = nhs.config.simulation_config(config_path)
+    except Exception as e:
+        logger.critical(
+            f"Failed to load configuration at {config_path} with exception {e}, terminating..."
+        )
+        exit(1)
+
+    logger.enable("nhs")
+    nhs.logging.config_logger(logger_config)
+
+    gnaf_path = os.path.join(data_config["gnaf_path"], gnaf_fname)
+    logger.info(f"Reading GNAF data from {gnaf_path}...")
+    houses_df = read_parquet(gnaf_path)
+    if not isinstance(houses_df, pl.LazyFrame):
+        logger.error(f"Failed to load GNAF file at {gnaf_path}, terminating...")
+        exit(1)
+
+    census_path = os.path.join(data_config["census_path"], census_fname)
+    logger.info(f"Reading census data from {census_path}...")
+    census = read_parquet(census_path)
+    if not isinstance(census, pl.LazyFrame):
+        logger.error(f"Failed to load census file at {census_path}, terminating...")
+        exit(1)
+
+    logger.info(f"Reading shapefile from {data_config['shapefile_path']}...")
+    area_polygons: gpd.GeoDataFrame = read_shapefile(
+        data_config["shapefile_path"], data_config["crs"]
+    )
```

Comment:
This is purely for the SA1 code now right?

Reply:
shapefile contains all area code, so not just the SA1 code but also name of the state and SA2, SA3, SA4 codes...